python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -122,9 +125,11 @@ def dist_inference(g, gnn_encoder, get_input_embeds, batch_size, fanout,
             sampler = dgl.dataloading.MultiLayerNeighborSampler(fanout_i, mask=edge_mask)
             dataloader = dgl.dataloading.DistNodeDataLoader(g, infer_nodes, sampler,
                                                             batch_size=batch_size,
-                                                            shuffle=True,
+                                                            shuffle=False,
                                                             drop_last=False)
 
+            start = time.time()
+            th.distributed.barrier()
```

Comment:
you should swap the two lines if you want to measure the time for distributed GNN inference.